modules/metastore/src/LifeCycle/LifeCycle.php

```diff
@@ -189,7 +189,7 @@ protected function distributionLoad(MetastoreItemInterface $data) {
 
     $downloadUrl = $metadata->data->downloadURL;
 
-    if (isset($downloadUrl) && !filter_var($downloadUrl, FILTER_VALIDATE_URL)) {
+    if (!empty($downloadUrl) && !filter_var($downloadUrl, FILTER_VALIDATE_URL)) {
       $resourceIdentifier = $downloadUrl;
```

Comment:
Also, $resourceIdentifier is only used once within that if block, so we could remove it and just pass $downloadUrl.